### PR TITLE
CBG-1139 Implement KV operations for gocb v2 (collection)

### PR DESCRIPTION
**Pull request by adamcfraser**
Includes updates to existing kv bucket tests to run against gocb v1 (bucket) and gocb v2 (collection).

http://uberjenkins.sc.couchbase.com:8080/job/sync-gateway-integration/530/

### CODE REVIEW

File: base/collection.go [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+
+	err = getRawResult.Content(&rv)
+	return rv, uint64(getRawResult.Cas()), err
+}
+
+func (c *Collection) GetAndTouchRaw(k string, exp uint32) (rv []byte, cas uint64, err error) {
+	getAndTouchOptions := &gocb.GetAndTouchOptions{
+		Transcoder: gocb.NewRawBinaryTranscoder(),
+	}
+	getAndTouchRawResult, err := c.Collection.GetAndTouch(k, expAsDuration(exp), getAndTouchOptions)
```
Comment (bbrks) [inline base/collection.go]: `err` is unused here. Is it possible getAndTouchRawResult is nil in this case and panic on line 161?

**Review by bbrks — CHANGES_REQUESTED** [review]
Just a couple of things

File: base/util_testing.go [hunk 2]
```diff
@@ -62,6 +62,40 @@ func GetTestBucket(t testing.TB) *TestBucket {
 	}
 }
 
+func GetTestBucketForDriver(t testing.TB, driver CouchbaseDriver) *TestBucket {
+	if driver == GoCBv2 {
+		// TODO: add GoCBv2 support to TestBucketPool.  Force use of sg_int_0 until
+		//  this is implemented
```
[the remark this answers is not in the document]
Comment (adamcfraser) [reply base/util_testing.go]: Makes sense, have reworked to reserve a TestBucket via GetTestBucketAndSpec and use the server/bucket information from there.

File: base/collection.go (hunk 1 above)
Comment (adamcfraser) [reply base/collection.go]: Agree - have fixed.

**Review by bbrks — APPROVED** [review]

**Pull request merged**